### PR TITLE
add install instructions to README

### PR DESCRIPTION
**Pull request by epwalsh**
This should clarify https://github.com/allenai/allennlp/issues/4155

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,4 +22,36 @@
...
+pip install allennlp-models==1.0.0rc3
+pip freeze | grep allennlp
+# > allennlp==1.0.0rc3
+# > allennlp-models==1.0.0rc3
+```
+
+### From source
+
+If you intend to install the models package from source, then you probably also want to [install `allennlp` from source](https://github.com/allenai/allennlp#installing-from-source).
+Once you have `allennlp` installed, run the following within the same Python environment:
```
Comment (nelson-liu) [inline README.md:51]: what version of allennlp should you be using with the latest models version? is it expected that models is updated lockstep with allennlp-core (i.e., always make sure you have the latest master for both)?

Comment (epwalsh) [reply README.md:51]: They are developed and tested side-by-side, so they should be kept up-to-date with each other. If you look at the GitHub Actions workflow for `allennlp-models`, it's always tested against the master branch of `allennlp`. Similarly, `allennlp` is always tested against the master branch of `allennlp-models`.

Comment (nelson-liu) [reply README.md:51]: That's good to know, thanks! Do you think that information should be added to the PR? I guess i'm imagining a scenario where I have to go to a previous commit of allennlp-model or allennlp-core, and your information would be helpful in figuring out what corresponding other hash i need.

Comment (epwalsh) [reply README.md:51]: Just added!

**Pull request merged**